src/util/component-types.ts

```diff
@@ -48,7 +48,8 @@ type TypesType<P> =
 	| { propName?: string | string[] }
 	| any; // TODO: figure out a type that works with inferred functional components
 
-interface ICreateClassComponentSpec<P, S> extends React.Mixin<P, S> {
```

Comment:
@jhsu I added the `any` default here because the type inference was doing really odd things for folks who didn't explicitly pass types to the generic.

Reply:
Actually turns out adding any types as defaults still allows the TS compiler to infer the types. Users will still need to manually add `<any, any>` to bypass most of the checking. I removed the defaults after more testing.